spec/rspec/support/differ_spec.rb

```diff
@@ -555,6 +556,75 @@ def inspect; "<BrokenObject>"; end
             expect(differ.diff(false, true)).to_not be_empty
           end
         end
+
+        unless RUBY_VERSION == '1.8.7' # We can't count on the ordering of the hash on 1.8.7, fuzzy matcher anything will continue broken on that version
+          describe "fuzzy matcher anything" do
+            it "outputs only key value pair that triggered diff, anything_key should absorb actual value" do
+              actual = { :fixed => "fixed", :trigger => "trigger", :anything_key => "bcdd0399-1cfe-4de1-a481-ca6b17d41ed8" }
+              expected = { :fixed => "fixed", :trigger => "wrong", :anything_key => anything }
+              diff = differ.diff(actual, expected)
+              expected_diff = dedent(<<-'EOD')
+                |
+                |@@ -1,4 +1,4 @@
+                | :anything_key => "bcdd0399-1cfe-4de1-a481-ca6b17d41ed8",
+                | :fixed => "fixed",
+                |-:trigger => "wrong",
+                |+:trigger => "trigger",
+                |
+              EOD
+              expect(diff).to be_diffed_as(expected_diff)
+            end
+
+            context "with nested hash" do
+              it "outputs only key value pair that triggered diff, anything_key should absorb actual value" do
+                actual = { :an_key => "dummy", :fixed => "fixed", :nested => { :anything_key => "bcdd0399-1cfe-4", :name => "foo", :trigger => "trigger" } }
+                expected = { :an_key => anything, :fixed => "fixed", :nested => { :anything_key => anything, :name => "foo", :trigger => "wrong" } }
+                diff = differ.diff(actual, expected)
+                expected_diff = dedent(<<-'EOD')
+                  |
+                  |@@ -1,4 +1,4 @@
+                  | :an_key => "dummy",
+                  | :fixed => "fixed",
+                  |-:nested => {:anything_key=>"bcdd0399-1cfe-4", :name=>"foo", :trigger=>"wrong"},
```

Comment:
I’m in the fence about this change. For nested arrays, doesn’t it make more sense to indicate that a certain key was `anything`?

Reply:
I don't have a strong opinion for nested arrays. Displaying that a certain key was anything for _nested_ arrays was the original behavior.

Reply:
I’m still not too familiar with this addition. Do you yhink it is possible to keep the effect of collapsed “an_key” which I undoubtedly like, but make no changes to the “anything_key”’s value in expected of a nested array?

Reply:
Yea, sure. If you checkout the [first commit](https://github.com/rspec/rspec-support/pull/596/commits/8c811465f38579ee72060a68656bcc306b7b2b1b) of this PR, you'll be on a state where there is no recursion, thus no further modifications on nested objects.

I have no problem at all to revert to that state. I don't have a strong opinion about morphing `anything` fuzzy matcher in nested objects. After all, when a nested object differs you get the entire object highlighted in red anyways, so I understand morphing deep nested values (like the one you are unfamiliar with: `:anything_key`) may not be worth to do, and it may be better to only keep the effect of collapsed first level keys like `"an_key"`.

But I don't know the answer to this, just want to give you options so you can choose a good answer. You have a lot more experience than I in this project :)